### DISCUSSION TITLE
Best Practices for Multi page application

### DISCUSSION THREAD

**Discussion by boogerlad** [slint-ui/slint#328] [Q&A]
Consider a setup wizard or something that requires an account (and subsequently, a login / signup). I'm aware the printer demo and the IOT dashboard feature multiple pages across several tabs, but I'm looking for how to implement sequential pages.

Secondly, how can a history of pages be maintained? For example, Android's back button.

**Comment by ogoffart** [comment]
It would be nice if we could have a nice way to do that. Idea and suggestion on how that could look like in .60 are welcome.

Currently the best one can do is to use a opaque `Rectangle { }` (with a `TouchArea` over the whole rectangle to make sure events)
that is to be shown (using a `if (condition) : ... ` construct) when that page is supposed to be shown.

An example of that is done in the printer demo.  First, you have tabs, and within the main tab, you have also extra screen:
https://github.com/sixtyfpsui/sixtyfps/blob/07c1504627806a213df7d08f7c810715d1ea7ead/examples/printerdemo/ui/home_page.60#L93
In this case, the screens are moved to the right when not visible. and there is a `current_subpage` property that decide which screen is active.

If you want to build a stack history of page, you would probably do that in the application logic.

This works, but it is true that a nicer / easier way would be great addition to our framework.
Maybe something like a `PageStack { ... }` widget, which would have one `active_page` property (an integer) and each sub element would have a `page_index` property injected, and maybe some way to define transitions on how the pages are animated when the active_page change.

**Reply by samyak-jain** [reply]
I agree. I think this kind of API design would be really helpful. Is there any open issue on this?

**Reply by MiKom** [reply]
> I agree. I think this kind of API design would be really helpful. Is there any open issue on this?

https://github.com/slint-ui/slint/issues/1722
